birdhouse/components/canarie-api/docker-compose-extra.yml

```diff
@@ -0,0 +1,45 @@
+version: "3.4"
+
+x-logging:
+  &default-logging
+  driver: "json-file"
+  options:
+    max-size: "50m"
+    max-file: "10"
+
+services:
+  canarie-api:
+    image: ${CANARIE_IMAGE}
+    container_name: canarie-api
+    volumes:
+      - canarie-data:/data/
+      - ./components/canarie-api/docker_configuration.py:/config/docker_configuration.py:ro
+      - ./components/canarie-api/entrypoint:/entrypoint:ro
+    environment:
+      CANARIE_API_CONFIG_FN: /config/docker_configuration.py
+      CANARIE_MONITORING_EXTRA_CONF_DIR: ${CANARIE_MONITORING_EXTRA_CONF_DIR}
+    restart: always
+    entrypoint: /entrypoint
+    logging: *default-logging
+
+  canarie-api-cron:
+    image: ${CANARIE_IMAGE}
+    container_name: canarie-api-cron
+    depends_on:
+      - canarie-api
+    # The volumes and environment should be the same as the canarie-api container except for the proxy-logs volume
+    volumes:
+      - canarie-data:/data/
+      - ./components/canarie-api/docker_configuration.py:/config/docker_configuration.py:ro
+      - ./components/canarie-api/entrypoint-cron:/entrypoint:ro
+      - proxy-logs:/logs/:ro  # aligns with the DATABASE["access_log"] value in docker_configuration.py.template
+    environment:
+      CANARIE_API_CONFIG_FN: /config/docker_configuration.py
+      CANARIE_MONITORING_EXTRA_CONF_DIR: ${CANARIE_MONITORING_EXTRA_CONF_DIR}
+    restart: always
+    entrypoint: /entrypoint
+    logging: *default-logging
+
+volumes:
+  canarie-data:
+  proxy-logs:
```

Comment:
Very much like this PR.  It's much cleaner this way that the canarie-api monitoring is separate from the proxy container.

I just wonder why are you using data-volume and not direct persistence on disk.  And if you really need to use data-volume, the current way we will lose the data on `./pavics-compose.sh down -v`.

To avoid losing data on `./pavics-compose.sh down -v`, use external data-volume like the monitoring component.

Reply:
A few reasons...

No other component is storing logs in a persistent manner. I'm not opposed to this but this could be a larger discussion about log retention that isn't related to this PR

> the current way we will lose the data on `./pavics-compose.sh down -v`

Sure, but are you bringing the stack down with the `-v` flag often? Personally, I never ever use the `-v` flag in production for that reason, and I'll delete a volume if needed manually.

I actually find it annoying to have to delete the externally created jupyterhub, prometheus, grafana, etc. volumes when I'm testing and want to start with a blank slate. I personally would actually like to see the externally created volumes become non-external (or a bind-mount from disk) for this reason.

Like I mentioned above, we don't currently have a defined log retention mechanism/policy. I'd love to add one, but maybe we can defer the discussion of how to manage logs when we discuss that.

Reply:
I think the way it is defined currently is the right way. 
If a platform maintainer wants to persist the logs at a specific location/drive/bind-mount, they simply need to add the relevant overrides to define the volume.
The flexibility is left up to the maintainer.

Reply:
> Sure, but are you bringing the stack down with the `-v` flag often?

I actually do use `-v` from time to time.  Otherwise, all the anonymous volume are not delete and next `up` new anonymous volumes will be created and the old anonymous volumes are only deleted with `docker volume prune --all`.

Here is an example of anonymous volume: https://github.com/bird-house/birdhouse-deploy/blob/80388564101f69d9ba93b532f41273611aa07c7a/birdhouse/components/finch/docker-compose-extra.yml#L21